packages/cli/src/eventbus/MessageEventBus/MessageEventBus.ts

```diff
@@ -0,0 +1,253 @@
+import { LoggerProxy, MessageEventBusDestinationOptions } from 'n8n-workflow';
+import { DeleteResult } from 'typeorm';
+import { EventMessageTypes } from '../EventMessageClasses/';
+import type { MessageEventBusDestination } from '../MessageEventBusDestination/MessageEventBusDestination.ee';
+import { MessageEventBusLogWriter } from '../MessageEventBusWriter/MessageEventBusLogWriter';
+import EventEmitter from 'node:events';
+import config from '@/config';
+import * as Db from '@/Db';
+import { messageEventBusDestinationFromDb } from '../MessageEventBusDestination/Helpers.ee';
+import uniqby from 'lodash.uniqby';
+import { EventMessageConfirmSource } from '../EventMessageClasses/EventMessageConfirm';
+import {
+	EventMessageAuditOptions,
+	EventMessageAudit,
+} from '../EventMessageClasses/EventMessageAudit';
+import {
+	EventMessageWorkflowOptions,
+	EventMessageWorkflow,
+} from '../EventMessageClasses/EventMessageWorkflow';
+import { isLogStreamingEnabled } from './MessageEventBusHelper';
+import { EventMessageNode, EventMessageNodeOptions } from '../EventMessageClasses/EventMessageNode';
+import {
+	EventMessageGeneric,
+	eventMessageGenericDestinationTestEvent,
+} from '../EventMessageClasses/EventMessageGeneric';
+
+export type EventMessageReturnMode = 'sent' | 'unsent' | 'all';
+
+class MessageEventBus extends EventEmitter {
+	static #instance: MessageEventBus;
+
+	isInitialized: boolean;
+
+	logWriter: MessageEventBusLogWriter;
+
+	destinations: {
+		[key: string]: MessageEventBusDestination;
+	} = {};
+
+	#pushIntervalTimer: NodeJS.Timer;
```

Comment:
I'm torn between staying consistent with the rest of the code using `private`, or using `#` to ensure private at runtime as well. 
The second one feels like the new semantic thing to do, but in my experience the debugging experience with private fields has been buggy and slow (likely due to the introspection calls having to go through the node's inspector api).
perhaps we could use `private pushIntervalTimer: NodeJS.Timer;` for now, and create a tech-debt ticket to switch to runtime-private fields?
WDYT?

Reply:
moving back to private for now